iXBRLViewerPlugin/__init__.py

```diff
@@ -117,14 +119,22 @@ def generateViewer(
     # extend XBRL-loaded run processing for this option
     if saveViewerDest is None:
         return
-    if (cntlr.modelManager is None 
+    if (cntlr.modelManager is None
         or len(cntlr.modelManager.loadedModelXbrls) == 0 
         or any(not mx.modelDocument for mx in cntlr.modelManager.loadedModelXbrls)):
-        cntlr.addToLog("No taxonomy loaded.")
+        cntlr.addToLog("No taxonomy loaded.", mesaageCode=ERROR_MESSAGE_CODE)
         return
     modelXbrl = cntlr.modelManager.modelXbrl
     if modelXbrl.modelDocument.type not in (Type.INLINEXBRL, Type.INLINEXBRLDOCUMENTSET):
-        cntlr.addToLog("No inline XBRL document loaded.")
+        cntlr.addToLog("No inline XBRL document loaded.", mesaageCode=ERROR_MESSAGE_CODE)
+        return
+    if not (isHttpUrl(viewerURL) or os.path.exists(viewerURL)):
```

Comment:
Won't this break with a relative URL?  A relative URL here would be resolved relative to the CWD, but what matters is whether it exists relative to the output directory.  

Further you might want to use an absolute path here, e.g. `/js/ixbrlviewer.js` which also won't resolve here, but may do once it's behind a web server and accessed via HTTP.

Perhaps we should make this a GUI-only error, and give CLI users more freedom to do stuff that may not work?

I'm also looking at the existing code from line 140, and am a bit confused.  If you give it a relative `viewerURL` it attempts to resolve it relative to the destination directory, and if it exists, it copies it from there.  In the simple case where the relative URL is in the same directory, you end up either copying it onto itself (if it's called `ixbrlviewer.js`) or duplicating it (if it's called anything else).

Maybe what's actually need here are separate, mutually exclusive `viewerURL` and `viewerPath` options: if you specify `viewerPath`, we attempt to copy the script, and reference it as `ixbrlviewer.js` (relative URL), if you specify `viewerURL` we don't attempt to copy the script or modify the URL, and it's the user's problem to make sure that the script is where it should be.

FWIW #614 adds an option to disable the copy script functionality, but I think I'd now prefer the above solution of separate parameters.

Reply:
This was a misunderstanding on my part. I wasn't aware that paths relative to the output directory were supported (the puppeteer tests failed because of this).

I investigated adding a `viewerPath` option, but that ran into the hurdle of how to make it optional while also providing the install directory script as a default value so users don't have to figure that out on their own. Ultimately I added a `--viewer-no-copy-script` option (also available in the GUI) to disable copying `--viewer-url` and validating the value. With the new configuration `--viewer-url` is confirmed to exist and downloaded/copied into the output directory (with the basename of the file) unless the `--viewer-no-copy-script` option is used. The negative argument name is unfortunate (I'm open to alternative names), but I think it provides the most intuitive behavior without needing to support 3 arguments. Let me know your thoughts.